sql/types/json_value.go

```diff
@@ -27,92 +27,63 @@ import (
 	"github.com/dolthub/go-mysql-server/sql"
 )
 
-// JSONValue is an integrator specific implementation of a JSON field value.
-type JSONValue interface {
-	// Unmarshall converts a JSONValue to a JSONDocument
-	Unmarshall(ctx *sql.Context) (val JSONDocument, err error)
-	// Compare compares two JSONValues. It maintains the same return value
-	// semantics as Type.Compare()
-	Compare(ctx *sql.Context, v JSONValue) (cmp int, err error)
-	// ToString marshalls a JSONValue to a valid JSON-encoded string.
-	ToString(ctx *sql.Context) (string, error)
-}
-
-// SearchableJSONValue is JSONValue supporting in-place access operations.
-// The query engine can utilize these optimized access methods improve performance
-// by minimizing the need to unmarshall a JSONValue into a JSONDocument.
-type SearchableJSONValue interface {
-	JSONValue
-
-	// Contains is value-specific implementation of JSON_Contains()
-	Contains(ctx *sql.Context, candidate JSONValue) (val interface{}, err error)
-	// Extract is value-specific implementation of JSON_Extract()
-	Extract(ctx *sql.Context, path string) (val JSONValue, err error)
-	// Keys is value-specific implementation of JSON_Keys()
-	Keys(ctx *sql.Context, path string) (val JSONValue, err error)
-	// Overlaps is value-specific implementation of JSON_Overlaps()
-	Overlaps(ctx *sql.Context, val SearchableJSONValue) (ok bool, err error)
-	// Search is value-specific implementation of JSON_Search()
-	Search(ctx *sql.Context) (path string, err error)
-}
-
-type MutableJSONValue interface {
+type StringifyJSON interface {
+	ToString() (string, error)
+}
+
+type MutableJSONV interface {
```

Comment:
(minor) Any reason not to leave the type name as `MutableJSONValue`? Seems more readable to leave it as `Value` instead of just `V`. Or... maybe just `MutableJSON` if you don't want to use the JSON Value term anymore?

Reply:
whoops this was an accident, meant to be `MutableJSON`